.github/workflows/python-publish.yml

```diff
@@ -39,13 +39,18 @@ jobs:
     - name: Build package
       run: python -m build --sdist
     - name: Publish package to TestPyPI
+      # upload to PyPI on every tag starting with 'v'
+      # if: github.event_name == 'push' && startsWith(github.event.ref, 'refs/tags/v')
+      # alternatively, to publish when a GitHub Release is created, use the following rule:
+      if: github.event_name == 'release' && github.event.action == 'published'
```

Comment:
BTW,  the trigger  for tags is not firing for 'v1.0' because it expects 'v1.0.x' as far I can tell. And there is the question if we shouldn't just trigger for the publish event.